proto/tendermint/liquidity/v1beta1/tx.proto

```diff
@@ -0,0 +1,207 @@
+syntax = "proto3";
+package tendermint.liquidity.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "cosmos_proto/coin.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+option go_package = "github.com/tendermint/liquidity/x/liquidity/types";
+
+// Msg defines the liquidity Msg service.
+service Msg {
+
+  // Submit create liquidity pool message.
+  rpc CreatePool(MsgCreatePool) returns (MsgCreatePoolResponse);
+
+  // Submit deposit to the liquidity pool batch.
+  rpc DepositWithinBatch(MsgDepositWithinBatch) returns (MsgDepositWithinBatchResponse);
+
+  // Submit withdraw from to the liquidity pool batch.
+  rpc WithdrawWithinBatch(MsgWithdrawWithinBatch) returns (MsgWithdrawWithinBatchResponse);
+
+  // Submit swap to the liquidity pool batch.
+  rpc Swap(MsgSwapWithinBatch) returns (MsgSwapWithinBatchResponse);
+}
+
+// MsgCreatePool defines an sdk.Msg type that supports submitting create liquidity pool
+//
+// See: https://github.com/tendermint/liquidity/blob/develop/x/liquidity/spec/04_messages.md
+message MsgCreatePool {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string pool_creator_address = 1 [(gogoproto.moretags) = "yaml:\"pool_creator_address\"",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "account address of the origin of this message",
+      example: "\"cosmos1e35y69rhrt7y4yce5l5u73sjnxu0l33wvznyun\"",
+      format: "sdk.AccAddress"
+    }];
+
+  // id of target pool type, only 1 is allowed on this version, Must match the value in the pool.
+  uint32 pool_type_id = 2 [(gogoproto.moretags) = "yaml:\"pool_type_id\"",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"1\"",
+      format: "uint32"
+  }];
+
+  // reserve coin pair of the pool to deposit
+  repeated cosmos.base.v1beta1.Coin deposit_coins = 4 [(gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"deposit_coins\"",
+    (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "[{\"denom\": \"denomX\", \"amount\": \"1000000\"}, {\"denom\": \"denomY\", \"amount\": \"2000000\"}]",
+      format: "sdk.Coins"
+    }];
+}
+
+// MsgCreatePoolResponse defines the Msg/CreatePool response type.
+message MsgCreatePoolResponse {}
+
+// `MsgDepositWithinBatch defines` an `sdk.Msg` type that supports submitting deposit request to the batch of the liquidity pool
+// Deposit submit to the batch of the Liquidity pool with the specified `pool_id`, `deposit_coins` for reserve
+// this requests are stacked in the batch of the liquidity pool, not immediately processed and
+// processed in the `endblock` at once with other requests.
```

Comment:
```suggestion
// `MsgDepositWithinBatch defines` an `sdk.Msg` type that supports submitting a deposit request to the batch of the liquidity pool
// The deposit is submitted with the specified `pool_id` and reserve `deposit_coins`.
// The deposit requests are stacked in the liquidity pool batch and are not immediately processed. 
//  Batch deposit requests are processed in the `endblock` at the same time as other requests.
```

Reply:
I have a question here, is "the batch of the liquidity pool" the same thing as "liquidity pool batch" or we need to find a way to describe batch processing?

Reply:
the light paper describes this very well, is this description I just rewrote accurate for batch execution? 

> Orders are accumulated in a liquidity pool for a pre-defined period. Orders are then added to the pool and executed at the end of the batch. 

if so, then what part of the batch execution are we describing here?

Reply:
> I have a question here, is "the batch of the liquidity pool" the same thing as "liquidity pool batch" or we need to find a way to describe batch processing?

I think we could go either way in my opinion.

> is this description I just rewrote accurate for batch execution?

Yes, what you rewrote is accurate for batch execution.  Deposit, withdraw, and swap requests are all processed in the `endblock`. 

> if so, then what part of the batch execution are we describing here?

I am not sure what you mean by "what part of the batch execution". Can you elaborate on this?